src/_guides/environment-declarations.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Configuring apps with compilation environment declarations
+description: >
+  Learn about using compilation environment declarations to customize application behavior.
+---
+
+Dart supports specifying compilation environment declarations when
+building or running an application.
+Compilation environment declarations are used to specify
+configuration values or options as key-value pairs
+that are accessed and evaluated at compile time.
+
+{{site.alert.note}}
+  This page uses "environment" and "environment declarations"
+  to refer to the compilation environment, not the OS environment
+  often referred to as "the environment" in software terminology.
+{{site.alert.end}}
+
+Your app can use the values of environment declarations
+to change its functionality or behavior.
+Dart compilers can take advantage of this to eliminate code
+no longer reachable due to the values specified, through "tree shaking".
```

Comment:
This comment isn't about changing the environment declaration values, but rather using them to make changes.

Also, the second sentence is "tree shaking", so I'm not sure this form of separation makes sense. I've decided to remove "tree shaking", since it doesn't really add anything unless we have a link destination for the term. 

I've pushed some changes, particularly for the second sentence to make it more clear when code can be removed. I continued to use the word "eliminate" though as that is the standard term used in dead code elimination.